source/solvers/heat_transfer.cc

```diff
@@ -782,6 +786,59 @@ HeatTransfer<dim>::attach_solution_to_output(DataOut<dim> &data_out)
                                this->present_solution,
                                heat_flux_postprocessors[m_id]);
     }
+
+  // Ouput the time average temperature and time average heat flux
+  if (simulation_parameters.post_processing.calculate_average_temp_and_hf &&
+      this->simulation_control->get_current_time() >
+        simulation_parameters.post_processing
+            .initial_time_for_average_temp_and_hf -
+          1e-6 * simulation_control->get_time_step())
```

Comment:
I would not do that. Just always output it. If the vector is not averaged yet it will have a value of zero
This is very convenient when you do animations. You can start the animations and when the averaging kicks in it will establish itself cleanly. I would always output it no matter what it will be minor.

Reply:
Just don't calculate it if it's below the averaging time.